src/biz/collab/ops.rs

```diff
@@ -498,6 +476,142 @@ pub async fn list_database_row_ids(
   Ok(db_rows)
 }
 
+pub async fn insert_database_row(
+  collab_storage: &CollabAccessControlStorage,
+  pg_pool: &PgPool,
+  workspace_uuid_str: &str,
+  database_uuid_str: &str,
+  uid: i64,
+  cell_value_by_id: HashMap<String, serde_json::Value>,
+) -> Result<String, AppError> {
+  let mut db_txn = pg_pool.begin().await?;
+
+  // get database types and type options
+  let (mut db_collab, db_body) =
+    get_database_body(collab_storage, workspace_uuid_str, database_uuid_str).await?;
+
+  let all_fields = db_body.fields.get_all_fields(&db_collab.transact());
+  let field_by_id = all_fields.iter().fold(HashMap::new(), |mut acc, field| {
+    acc.insert(field.id.clone(), field.clone());
+    acc
+  });
+  let type_option_reader_by_id = type_option_writer_by_id(&all_fields);
+  let field_by_name = field_by_name_uniq(all_fields);
+
+  let new_db_row_id = uuid::Uuid::new_v4().to_string();
```

Comment:
we might have a function call: gen_row_id()

Reply:
I've used this